### PR TITLE
Delete Org (Test Review)

### PR DESCRIPTION
**Pull request by anusuyar**

### CODE REVIEW

**Comment by anusuyar** [comment]
This change is [Reviewable](https://reviewable.io/reviews/anusuyar/pyvcloud/4)

**Comment by anusuyaceg** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[pyvcloud/vcd/system.py, line 67 at r1](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-KzWyWo8VvI8wV2f687w:-KzWyWo9yf0n5l7oxRD_:b-57fh1p) ([raw file](https://github.com/anusuyar/pyvcloud/blob/81f849131da5430d3543b215370689454b9e4092/pyvcloud/vcd/system.py#L67)):*
> ```Python
>         raise Exception('org \'%s\' not found' % org_name)
> 
>     def delete_org(self, org_name, force=None, recursive=None):
> ```

Swap force and recursive?

---

*[tests/vcd_system.py, line 16 at r1](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-KzWyctDLfUGGfFuxm-U:-KzWyctDLfUGGfFuxm-V:b1x22xt) ([raw file](https://github.com/anusuyar/pyvcloud/blob/81f849131da5430d3543b215370689454b9e4092/tests/vcd_system.py#L16)):*
> ```Python
>         assert org.get('name') == "TestOrgName"
> 
>     def test_delete_org(self):
> ```

Test review

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by anusuyar** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[pyvcloud/vcd/system.py, line 67 at r1](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-KzWyWo8VvI8wV2f687w:-KzWz7n8Hm-cFFRMxObJ:bb7m07t) ([raw file](https://github.com/anusuyar/pyvcloud/blob/81f849131da5430d3543b215370689454b9e4092/pyvcloud/vcd/system.py#L67)):*
<details><summary><i>Previously, anusuyaceg wrote…</i></summary><blockquote>

Swap force and recursive?
</blockquote></details>

okay

---

*[tests/vcd_system.py, line 16 at r1](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-KzWyctDLfUGGfFuxm-U:-KzWzARIxLTPmdrY3ITR:bb7m07t) ([raw file](https://github.com/anusuyar/pyvcloud/blob/81f849131da5430d3543b215370689454b9e4092/tests/vcd_system.py#L16)):*
<details><summary><i>Previously, anusuyaceg wrote…</i></summary><blockquote>

Test review
</blockquote></details>

okay

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

File: pyvcloud/vcd/system.py [hunk 1]
```diff
@@ -51,3 +51,30 @@ def create_org(self, org_name, full_org_name):
...
+        Delete an organization.
+        :param org_name: name of the org to be deleted.
+        :param force: pass force=True  along with recursive=True to remove an
+        organization and any objects it contains, regardless of their state.
+        :param recursive: pass recursive=True  to remove an organization
+        and any objects it contains that are in a state that normally allows.
+        removal.
+         """  # NOQA
+        org = self.get_org(org_name)
+        org_href = org.get('href').replace('/api/org/',
```
Comment (sahithi) [inline pyvcloud/vcd/system.py]: I noticed that System object has admin_href variable. Please check if we can use it instead of replace call.

**Review by sahithi — CHANGES_REQUESTED** [review]
Please address comments

**Comment by anusuyar** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pyvcloud/vcd/system.py, line 78 at r1](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-KzWyMpJePR_4bOyNy9M-r1-78:-KzzUBpBeFHlCIk8-NNx:b-3lmn8l) ([raw file](https://github.com/anusuyar/pyvcloud/blob/b033de670f57ee0f4ee7f312893a5d01629ea990/pyvcloud/vcd/system.py#L78)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

I noticed that System object has admin_href variable. Please check if we can use it instead of replace call.
</blockquote></details>

admin_href denotes the system admin href. Here we need the href of the org we need to delete. I don't think we can use admin_href here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by anusuyar** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pyvcloud/vcd/system.py, line 78 at r1](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-KzWyMpJePR_4bOyNy9M-r1-78:-KzzWZP7RnNbjymgMd8o:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/b033de670f57ee0f4ee7f312893a5d01629ea990/pyvcloud/vcd/system.py#L78)):*
<details><summary><i>Previously, anusuyar wrote…</i></summary><blockquote>

admin_href denotes the system admin href. Here we need the href of the org we need to delete. I don't think we can use admin_href here.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by rocknes** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[pyvcloud/vcd/system.py, line 78 at r1](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-KzWyMpJePR_4bOyNy9M-r1-78:-L-8Q7wALbPMkVJRMHNf:b-8zijrc) ([raw file](https://github.com/anusuyar/pyvcloud/blob/b033de670f57ee0f4ee7f312893a5d01629ea990/pyvcloud/vcd/system.py#L78)):*
<details><summary><i>Previously, anusuyar wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Test comment.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by rocknes** [comment]
*[pyvcloud/vcd/system.py, line 42 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-L-8mABMIaueAzaucZeW:-L-8mABMIaueAzaucZeX:b-z0fc89) ([raw file](https://github.com/anusuyar/pyvcloud/blob/fb96d21aced7d38d413b3d0da36581ce6704b451/pyvcloud/vcd/system.py#L42)):*
> ```Python
>         :param full_org_name: The fullname of the organization.
>         :return: (AdminOrgType) Created org object.
>          """  # NOQA
> ```

extra space in front of "'"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by rocknes** [comment]
Review status: 0 of 2 files reviewed at latest revision, 5 unresolved discussions.

---

*[pyvcloud/vcd/system.py, line 60 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-L-8mHGWsovHvWi3Dh0V:-L-8mHGWsovHvWi3Dh0W:bobtvlj) ([raw file](https://github.com/anusuyar/pyvcloud/blob/fb96d21aced7d38d413b3d0da36581ce6704b451/pyvcloud/vcd/system.py#L60)):*
> ```Python
>         :param org_name: name of the org to be retrieved.
>         :return: Org record.
>          """  # NOQA
> ```

Extra space in front of "'"

---

*[pyvcloud/vcd/system.py, line 62 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-L-8oUvO6qe6KxfJ0_14:-L-8oUvO6qe6KxfJ0_15:bt4gu91) ([raw file](https://github.com/anusuyar/pyvcloud/blob/fb96d21aced7d38d413b3d0da36581ce6704b451/pyvcloud/vcd/system.py#L62)):*
> ```Python
>          """  # NOQA
>         orgs = self.client.get_org_list()
>         for org in [o for o in orgs.Org if hasattr(orgs, 'Org')]:
> ```

Probably we should extract this as a separate function in client.py  viz. get_org_by_name(). Since this exact piece of code is being repeated in vcd-cli too.

---

*[tests/vcd_system.py, line 13 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-L-8rbPOw6hprJDfASwA:-L-8rbPOw6hprJDfASwB:b-c5bxq8) ([raw file](https://github.com/anusuyar/pyvcloud/blob/fb96d21aced7d38d413b3d0da36581ce6704b451/tests/vcd_system.py#L13)):*
> ```Python
>         sys_admin = self.client.get_admin()
>         system = System(self.client, admin_resource=sys_admin)
>         org = system.create_org("TestOrgName", "orgFullName")
> ```

Please don't hardcode values. self.config['vcd'][<something>]

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by rocknes** [comment]
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by anusuyar** [comment]
Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[pyvcloud/vcd/system.py, line 60 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-L-8mHGWsovHvWi3Dh0V:-L-9LLlWdcbfvHGUA50w:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/fb96d21aced7d38d413b3d0da36581ce6704b451/pyvcloud/vcd/system.py#L60)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

Extra space in front of "'"
</blockquote></details>

Done.

---

*[pyvcloud/vcd/system.py, line 62 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-L-8oUvO6qe6KxfJ0_14:-L-9LSmJDNUY5bWnCJ_I:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/fb96d21aced7d38d413b3d0da36581ce6704b451/pyvcloud/vcd/system.py#L62)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

Probably we should extract this as a separate function in client.py  viz. get_org_by_name(). Since this exact piece of code is being repeated in vcd-cli too.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by anusuyar** [comment]
Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[tests/vcd_system.py, line 13 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-L-8rbPOw6hprJDfASwA:-L-9MZj6WpQY2qDgVN7q:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/fb96d21aced7d38d413b3d0da36581ce6704b451/tests/vcd_system.py#L13)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

Please don't hardcode values. self.config['vcd'][<something>]
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by anusuyaceg** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r1.
Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-:-L-9OGB0McIZR-xIqaiX:bnfp4nl)*

**Comment by anusuyar** [comment]
Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[pyvcloud/vcd/system.py, line 42 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-L-8mABMIaueAzaucZeW:-L-9UMvCBuD0LyA0Y-iq:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/fb96d21aced7d38d413b3d0da36581ce6704b451/pyvcloud/vcd/system.py#L42)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

extra space in front of "'"
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by rocknes** [comment]
Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by rocknes** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-:-L-9fhtPDfdN3qrUkJaF:bnfp4nl)*

**Comment by sahithi** [comment]
Address open comment

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[pyvcloud/vcd/system.py, line 78 at r1](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-KzWyMpJePR_4bOyNy9M-r1-78:-L-CxgKghoQWCPOLniqa:bmstuj7) ([raw file](https://github.com/anusuyar/pyvcloud/blob/b033de670f57ee0f4ee7f312893a5d01629ea990/pyvcloud/vcd/system.py#L78)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

Test comment.
</blockquote></details>

Anusuya, sorry for the late comment. Take a look at the CreateOrg method
      "if self.admin_resource is None: "
Once the admin_resource is set, all further queries would have /api/admin/ in the url.
org = self.client.get_org_by_name(org_name) - this would return org whose href looks like .../api/admin/org/....
That is my understanding. Please do a quick test and see if it works. If it doesn't, consider my approval for the current change.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-:-L-CzWTIqPJV8_ifXWrp:b43o3q2)*

**Comment by sahithi** [comment]
Reviewed 4 of 4 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4)*

**Comment by sahithi** [comment]
LGTM

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-:-L-DoMUVmqUIhK3tdfJ0:bb6ywmx)*

**Comment by sahithi** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/4#-:-L-Doj7LaC9aqFmaKcje:bnfp4nl)*

**Pull request merged**